File: LZW/LZW_encode.py
# -*- coding: utf-8 -*-
with open('LZW_toEncode.txt', 'r') as f:
	s = f.read()
# print('File contains:\n-----\n{}\n-----'.format(s))
d = [chr(i) for i in range(256)]

stack = ''
flow = []

for c in s:
	stack += c
	if stack not in d:
		flow.append(d.index(stack[:-1]))
		d.append(stack)
		stack = c

if stack:
	flow.append(d.index(stack))

with open('LZW_Encoded.txt', 'w') as F:
	F.write('\n'.join(map(str, flow)))

File: LZW/LZW_encode.cpp
#include <iostream>
#include <fstream>
#include <map>
#include <string>
#include <ctime>

using namespace std;

int main()
{
	clock_t timeStart = clock();
	ifstream fi("LZW_toEncode.txt");

	if (fi.is_open()) {
		ofstream fo("LZW_Encoded.txt");

		if (fo.is_open()) {
			char c;
			string stack;
			string newstack;
			map<string, int> d;
			for (int i=0; i<256; ++i) {
				d[string(1, i)] = i;
			}
			int last = 256;

			while (fi.get(c)) {
				newstack = stack + c;
				if (d.find(newstack) == d.end()) {
					fo << d[stack] << " ";
					d[newstack] = last++;
					newstack = c;
				}
				stack = newstack;
			}
			fo << d[stack]; // BEWARE! :O
		} else {
			cout << "Unable to open output file :C" << endl;
		}
		fi.close();
	} else {
		cout << "Unable to open input file :c" << endl;
	}
	cout << "Done.\nTime: " << (double)(clock() - timeStart)/CLOCKS_PER_SEC*1000. << " ms" << endl;
	return 0;
}
